main.go

```diff
@@ -23,14 +43,32 @@ func main() {
 			content = append(content, '\n')
 		}
 
-		if bytes.Compare(checkEquality, content) != 0 {
-			_, _ = os.Stdout.Write(content)
-		}
+		if *uniqueParameter {
+			if bytes.Compare(checkEquality, []byte("")) != 0 {
+				if bytes.Compare(checkEquality, test) == 0 {
+					_, _ = os.Stdout.Write(content)
+					checkEquality = []byte("")
+					continue
 
-		if readErr == io.EOF {
-			break
+				} else {
+					if bytes.Compare(checkEquality, content) != 0 {
+						_, _ = os.Stdout.Write(checkEquality)
+					}
+				}
+			}
+			if readErr == io.EOF {
+				_, _ = os.Stdout.Write(content)
+				break
+			}
+		} else {
+			if bytes.Compare(checkEquality, content) != 0 {
+				_, _ = os.Stdout.Write(content)
+				if readErr == io.EOF {
+					break
+				}
+			}
 		}
-
+		test = checkEquality
```

Comment:
Why don't you check for EOF in all cases now?

Reply:
And this one?

https://github.com/Al2Klimov/trainee-go-uniq/pull/12/files#diff-7ddfb3e035b42cd70649cc33393fe32cR36